### PR TITLE
Improved some default parameters

### PR DESCRIPTION
**Pull request by g3force**

### CODE REVIEW

**Comment by g3force** [comment]
#139 needs to be merged first

File: src/app/plugins/plugin_detect_balls.h [hunk 1]
```diff
@@ -74,11 +74,11 @@ friend class PluginDetectBalls;
 
   _settings=new VarList("Ball Detection");
 
-  _settings->addChild(_max_balls = new VarInt("Max Ball Count",10));
+  _settings->addChild(_max_balls = new VarInt("Max Ball Count",20));
   _settings->addChild(_color_label = new VarString("Ball Color","Orange"));
 
   _settings->addChild(_filter_general = new VarList("Ball Properties"));
-    _filter_general->addChild(_ball_z_height = new VarDouble("Ball Z-Height", 30.0));
+    _filter_general->addChild(_ball_z_height = new VarDouble("Ball Z-Height", 21.5));
```
Comment (g3force) [inline src/app/plugins/plugin_detect_balls.h]: I'm still not sure if the 30mm were intended.
The ball has a radius of 21.5mm. When looking from above, I see the full shape of the ball, which is at a height of one radius.

Comment (g3force) [reply src/app/plugins/plugin_detect_balls.h]: I've experimented with this value on my field and it turned out that 30mm actually seems to be good. I still do not fully understand, why, though.

Comment (joydeep-b) [reply src/app/plugins/plugin_detect_balls.h]: Yes, we have consistently observed the same, which is why it is so. Can you add a comment to the effect? Let's prevent people in the future try to set it back. ;-)
I suspect it is partially due to the ball sinking into the carpet, and partially due to bias in the geometry calibrator.

File: src/shared/util/camera_calibration.cpp [hunk 2]
```diff
@@ -745,12 +745,9 @@ CameraParameters::AdditionalCalibrationInformation::
     }
 
   initial_distortion = new VarDouble("initial distortion", 1.0);
-  camera_height = new VarDouble("camera height", 4000.0);
-  line_search_corridor_width = new VarDouble("line search corridor width",
-                                             280.0);
+  line_search_corridor_width = new VarDouble("line search corridor width", 280.0);
```
Comment (joydeep-b) [inline src/shared/util/camera_calibration.cpp]: This change might conflict with the cleanup PR #139 
Recommend you merge #139, then pull into this branch first.

File: src/app/plugins/plugin_detect_balls.cpp [hunk 3]
```diff
@@ -248,7 +248,9 @@ ProcessResult PluginDetectBalls::process ( FrameData * data, RenderOptions * opt
             for (int r = 0; r < robots_n; r++) {
               const SSL_DetectionRobot & robot = robots->Get(r);
               if (robot.confidence() > 0.0) {
-                if ((sq((double)(robot.x())-(double)(field_pos.x)) + sq((double)(robot.y())-(double)(field_pos.y))) < near_robot_dist_sq) {
+                vector3d field_on_bot_pos_3d;
+                camera_parameters.image2field ( field_on_bot_pos_3d, pixel_pos, robot.height());
```
Comment (joydeep-b) [inline src/app/plugins/plugin_detect_balls.cpp]: Why is this necessary? It is already done in lines 217-220.

File: src/shared/util/camera_calibration.cpp (hunk 2 above)
Comment (g3force) [reply src/shared/util/camera_calibration.cpp]: I already noted this in a comment above ;) I based this branch on the cleanup branch, so that there will be no conflict ;)

File: src/app/plugins/plugin_detect_balls.cpp (hunk 3 above)
Comment (g3force) [reply src/app/plugins/plugin_detect_balls.cpp]: the difference is, that here I transform the blob to robot height.
Because if e.g. a magenta blob is falsely detected as orange und thus inside the bot pattern, it is at robot height. The filter did not behave correctly before.

File: src/app/plugins/plugin_detect_balls.h (hunk 1 above)
Comment (g3force) [reply src/app/plugins/plugin_detect_balls.h]: done.

File: src/app/plugins/plugin_detect_balls.cpp (hunk 3 above)
Comment (joydeep-b) [reply src/app/plugins/plugin_detect_balls.cpp]: Ah, I see. Yes, makes sense.
